src/App.tsx

```diff
@@ -1,14 +1,44 @@
+/* eslint-disable import/no-cycle */
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import { getTodos } from './api';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filter, setFilter] = useState('all');
+  const [query, setQuery] = useState('');
+  const [selectedTodoId, setSelecetedTodoId] = useState(0);
+  const [isLoading, setIsLoading] = useState(true);
+
+  const selectedTodo = todos.find(todo => todo.id === selectedTodoId);
+  const filteredTodos = todos.filter(todo => todo.title.toLowerCase().includes(query.toLowerCase()))
+    .filter(todo => {
+      switch (filter) {
+        case 'active':
+          return !todo.completed;
+
+        case 'completed':
+          return todo.completed;
+
+        default:
+          return todo;
+      }
+    });
+
+  useEffect(() => {
+    getTodos()
+      .then((todosFromServer: Todo[]) => setTodos(todosFromServer))
+      .finally(() => setIsLoading(false));
+  }, [todos]);
```

Comment:
wrong dependency: it caused infinity loop (todos have changed => run useEffect => fetch todos => setTodos => todos have changed - check `network` tab in devTools).
you can add empty dependencies array or add `[setIsLoading, setTodos]` - both cases will have the same result